include/game/multiplayer/multiplayer_event_mgr.h

```diff
@@ -8,6 +8,84 @@
 
 #include "game/sa1_sa2_shared/globals.h"
 
+// RoomEvent types
+#define ROOMEVENT_TYPE_NONE                  0
+#define ROOMEVENT_TYPE_PLATFORM_CHANGE       1
+#define ROOMEVENT_TYPE_ITEMBOX_BREAK         2
+#define ROOMEVENT_TYPE_ENEMY_DESTROYED       3
+#define ROOMEVENT_TYPE_PLAYER_RING_LOSS      4
+#define ROOMEVENT_TYPE_MYSTERY_ITEMBOX_BREAK 5
+#define ROOMEVENT_TYPE_ITEMEFFECT_APPLIED    6
+#define ROOMEVENT_TYPE_REACHED_STAGE_GOAL    7
+#define ROOMEVENT_TYPE_UNKNOWN               8
+
+// Common RoomEvent variables
+#define ROOMEVENT_BASE u8 type
+
+// RoomEvent opaque struct
+struct RoomEvent {
+    ROOMEVENT_BASE;
+
+    u8 opaque[7];
+}; /* 0x8 */
+
+// RoomEvent variable structs
+struct RoomEvent_PlatformChange {
+    ROOMEVENT_BASE;
+
+    u8 x;
+    u8 y;
+    u8 id;
+    u8 action;
+} platformChange;
+
+struct RoomEvent_ItemBoxBreak {
+    ROOMEVENT_BASE;
+
+    u8 x;
+    u8 y;
+    u8 id;
+} itemBoxBreak;
+
+struct RoomEvent_EnemyDestroy {
+    ROOMEVENT_BASE;
+
+    u8 x;
+    u8 y;
+    u8 id;
+} enemyDestroy;
+
+struct RoomEvent_RingLoss {
+    ROOMEVENT_BASE;
+
+    u8 ringCount;
+};
+
+struct RoomEvent_MysteryItemBoxBreak {
```

Comment:
We've actually been trying to move towards typedef structs, since the word `struct` can be a be redundant. Don't need to make this change now though if you don't have time.